webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -1,91 +1,233 @@
         super(props);
         this.state = {
-
+            tagMenu: false,
+            error: false,
+            addNewTags: [],
+            eventLanguages: []
         }
     };
 
-
     componentDidMount() {
-        this.fetchForm()
-        this.fetchData()
+        this.fetchForm();
+        this.fetchData();
+        this.fetchTags();
+        this.fetchEvent();
     }
 
+    // Data Functions
+    // Fetch Event Details
+    fetchEvent() {
+        eventService.getEvent().then(response => {
+            this.setState({
+                eventLanguages: ["en", "fr"]
+            })
+        });
+    }
+
+
     // Fetch Form
     fetchForm() {
         applicationFormService.getForEvent(this.props.event.id).then(response => {
             this.setState({
                 applicationForm: response.formSpec
             })
-        })
-
+        });
     }
 
 
     // Fetch Data
     fetchData() {
         let params = {
             id: this.props.match.params.id
-        }
+        };
         fetchResponse(params).then(response => {
             this.setState({
                 applicationData: response
             })
-        })
-
-
+        });
     }
 
 
+    // Misc FUnctions
     // Go Back
     goBack() {
         this.props.history.goBack();
     }
 
 
+    error(error) {
+        this.setState({
+            error: error
+        })
+    }
+
+
+    // Tag Functions
+    // Populate tag list
+    fetchTags() {
+        tagList.list().then(response => {
+            this.setState({
+                tagList: response
+            })
+        });
+    }
+
+
+    // Toggle Tag Menu
+    toggleTags(list) {
+        this.setState({
+            tagMenu: list ? false : true
+        })
+    }
+
+
+    // Post Tag
+    postTag(tag, type) {
+        type == "responseTag" ? this.postResponseTag(tag) : this.postTagList(tag);
+    }
+
+
+
+    // Post Response API
+    postResponseTag(tag) {
+        const updateApplicationData = this.state.applicationData;
+        const updateTagList = this.state.tagList;
+
+        tagResponse.post(tag).then(response => {
+            if (response.status == 201) {
+                let getTag = this.state.tagList.filter(tag => {
+                    if (tag.id == response.tag_id) {
+                        return tag
+                    }
+                });
+                updateApplicationData.tags.push({ "headline": getTag[0].name, "id": getTag[0].id });
+                updateTagList.splice(getTag, 1);
+
+                this.setState({
+                    applicationData: updateApplicationData,
+                });
+            };
+        }).catch((response) => {
+            this.error(response.message)
+        });
+    }
+
+
+
+    // Post Tag List API
+    postTagList(tags) {
+        let updateApplicationData = this.state.applicationData;
+        let updateEventLanguages = this.state.eventLanguages;
+        let newTags = Object.values(tags);
+        let filterTags = tags;
```

Comment:
These can all be const

Reply:
I am modifying the values of updateApplicationData and updateEventLanguages, are you sure these can be const's?